src/Symfony/Component/RateLimiter/Limit.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+/**
+ * @author Valentin Silvestre <vsilvestre.pro@gmail.com>
+ *
+ * @experimental in 5.2
+ */
+final class Limit
+{
+    private $availableTokens;
+    private $retryAfter;
+    private $accepted;
+    private $strategy;
+    private $metadata;
+
+    public function __construct(int $availableTokens, \DateTimeImmutable $retryAfter, bool $accepted, string $strategy, array $metadata = [])
+    {
+        $this->availableTokens = $availableTokens;
+        $this->retryAfter = $retryAfter;
+        $this->accepted = $accepted;
+        $this->strategy = $strategy;
+        $this->metadata = $this->configureMetadata($metadata);
+    }
+
+    public function isAccepted(): bool
+    {
+        return $this->accepted;
+    }
+
+    public function getRetryAfter(): \DateTimeImmutable
+    {
+        return $this->retryAfter;
+    }
+
+    public function getRemainingTokens(): int
+    {
+        return $this->availableTokens;
+    }
+
+    public function getStrategy(): string
+    {
+        return $this->strategy;
+    }
+
+    public function getMetadata(): array
+    {
+        return $this->metadata;
+    }
+
+    private function configureMetadata(array $metadata): array
```

Comment:
I'd suggest inlining this method, it's used once in the constructor
but do we really need this field? It duplicates the other ones and is not used anywhere if I read the PR correctly?

Reply:
@wouterj wdyt about this? What should we keep or not?

Reply:
@wouterj wdyt about this? What should we keep or not?

Reply:
Agreed with Nicolas. Let's not duplicate information and only use metadata to store "arbitrary usefull information" per algorithm.